pkg/cluster/config/v1alpha2/conversion.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by conversion-gen. DO NOT EDIT.
+
+package v1alpha2
+
+import (
+	conversion "k8s.io/apimachinery/pkg/conversion"
+	config "sigs.k8s.io/kind/pkg/cluster/config"
+)
+
+func Convert_v1alpha2_Config_To_config_Cluster(in *Config, out *config.Cluster, s conversion.Scope) error {
+	for _, node := range in.Nodes {
+		// TODO(bentheelder): if v1alpha3 does re-gain per-node patches, drop
+		// this snippet
+		// Copy config patches from nodes to the cluster level, since we only
+		// generate a cluster-wide config anyhow
+		for _, patch := range node.KubeadmConfigPatches {
+			out.KubeadmConfigPatches = append(out.KubeadmConfigPatches, patch)
+		}
+		for _, patch := range node.KubeadmConfigPatchesJSON6902 {
+			out.KubeadmConfigPatchesJSON6902 = append(out.KubeadmConfigPatchesJSON6902, patch)
+		}
+
+		// skip now-implicit external load balancers
+		if node.Role == ExternalLoadBalancerRole {
+			continue
```

Comment:
I don't think we can just skip this altogether here - it will mean that a conversion from v1alpha2->internal->v1alpha3->internal->v1alpha2 will not be bijective, and will produce different results.

This is an important gotcha of writing conversions - in practice, I'm not sure what the effect of *not* doing it is, but the apimachinery assumes that this *is* the case, so we are breaking the rules here 🙄 

Instead here, we would usually store some additional information on annotations on the new resource, so that we can convert *back* from that format.

Reply:
Er so maybe this is the wrong place to do this but effectively I do want to drop this. Previously you needed to specify this and now it is implicit.

We don't support roundtripping properly because we did actually break between versions which is only allowed because the types are alpha by my understanding.

We also don't ever convert backwards yet. Only to internal.

Reply:
something similar was discussed in one of my comments earlier.
we are indeed breaking the rountrip rules a little here, so it's a question if we want the full rountrip support now, or close our eyes for the alpha2 -> alpha3 transition, but consider the proper conversations for say alpha3->next.

cc @fabriziopandini

Reply:
yeah, I think roundtrip support is important long term, but we're intentionally breaking compatibility between these now in alpha, you cannot express roundtrip because of this. we could have just deleted all old alphas.

Reply:
beta and onwards versions should be round trippable without issue.